bitnami/kafka/templates/svc.yaml

```diff
@@ -22,21 +22,10 @@ spec:
       port: {{ .Values.service.port }}
       protocol: TCP
       targetPort: kafka
-      {{- if and (or (eq .Values.service.type "NodePort") (eq .Values.service.type "LoadBalancer")) (not (empty .Values.service.nodePorts.kafka)) }}
-      nodePort: {{ .Values.service.nodePorts.kafka }}
+      {{- if and (or (eq .Values.service.type "NodePort") (eq .Values.service.type "LoadBalancer")) .Values.service.nodePort }}
+      nodePort: {{ .Values.service.nodePort }}
       {{- else if eq .Values.service.type "ClusterIP" }}
       nodePort: null
       {{- end }}
-    {{- if .Values.auth.ssl }}
-    - name: tls-kafka
-      port: {{ .Values.service.sslPort }}
-      protocol: TCP
-      targetPort: kafka-ssl
-      {{- if and (or (eq .Values.service.type "NodePort") (eq .Values.service.type "LoadBalancer")) (not (empty .Values.service.nodePorts.ssl)) }}
-      nodePort: {{ .Values.service.nodePorts.ssl }}
-      {{- else if eq .Values.service.type "ClusterIP" }}
-      nodePort: null
-      {{- end }}
-    {{- end }}
```

Comment:
This doesn't convince me yet. I understand your point and I agree with you that "complex configurations" should be addressed by customising the `listeners` and `advertisedlisteners` parameters.
That said, with these changes I can only expose one port. This is fine for the "default behaviour" but it's not valid for the "complex configurations" where I can have use `SASL_SSL` for external access and `SSL` for inter-broker communication. In that case I need to expose 2 ports.
So.. if you want to deprecate the `service.sslPort` parameter (I"m fine with it) we should add an alternative (e.g. sth like `service.extraPorts`) to add more ports to expose when needed.